### PR TITLE
examples: tproxy: Add BPF tproxy example

### PR DESCRIPTION
**Pull request by danobi**
This example application implements the classic iptables TPROXY target
using BPF. This is a nice example b/c it demonstrates a non-trivial BPF +
networking use case using libbpf-rs.

Signed-off-by: Daniel Xu <dxu@dxuuu.xyz>

### CODE REVIEW

**Comment by anakryiko** [comment]
this looks short and simple enough to be part of libbpf-bootstrap maybe?

**Comment by danobi** [comment]
Yeah, could be libbpf-bootstrap.

But just wanna add that one thing I really like about rust ecosystem is that examples in a project are generally in a standard location. libbpf-rs/libbpf-cargo doesn't strictly follow that pattern b/c we need to run build scripts per-example, but at least we have the `examples/` directory in the root.

For example, when I was developing `below` and using `cursive` (TUI crate), I found it incredibly helpful that they had a large directory of examples: https://github.com/gyscos/cursive/tree/main/cursive/examples . So I expect that libbpf-rs users might find a healthy examples/ directory here helpful as well.

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[*.[ch]]
```
Comment (insearchoflosttime) [inline .editorconfig:1]: is "root = true" also needed?

File: examples/tproxy/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+[package]
+name = "tproxy"
+version = "0.1.0"
+authors = ["Daniel Xu <dxu@dxuuu.xyz>"]
+edition = "2021"
```
Comment (insearchoflosttime) [inline examples/tproxy/Cargo.toml:5]: 2022? :)

Comment (danobi) [reply examples/tproxy/Cargo.toml:5]: Doesn't look like there's a 2022 edition yet: https://doc.rust-lang.org/edition-guide/introduction.html

File: .editorconfig (hunk 1 above)
Comment (danobi) [reply .editorconfig:1]: Looks like not needed but helpful.

**Pull request merged**